src/platform/nrfconnect/OTAImageProcessorImpl.h

```diff
@@ -19,34 +19,23 @@
 #include <lib/core/OTAImageHeader.h>
 #include <lib/support/Span.h>
 #include <platform/OTAImageProcessor.h>
+#include <platform/nrfconnect/ExternalFlashManager.h>
 
 namespace chip {
 
 class OTADownloader;
 
 namespace DeviceLayer {
 
-class FlashHandler
-{
-public:
-    enum class Action : uint8_t
-    {
-        WAKE_UP,
-        SLEEP
-    };
-    virtual ~FlashHandler() {}
```

Comment:
The virtual has been added here to suggest that this class could have been overloaded to handle not only QSPI driven external flash (the default one) but also other cases (like SPI or even different bus). The `GetFlashHandler` could then return the specialized object which could be injected to OTAOImageProcessorImpl. What do you think about persisting this?